tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -194,6 +194,24 @@ public function testFetchAllColumn() : void
         self::assertEquals([1], $stmt->fetchAll(FetchMode::COLUMN));
     }
 
+    public function testGetIterator() : void
+    {
+        $query = $this->connection->getDatabasePlatform()
+            ->getDummySelectSQL('1');
+
+        $qcp = new QueryCacheProfile(0, 0, new ArrayCache());
+
+        $stmt = $this->connection->executeCacheQuery($query, [], [], $qcp);
+        $stmt->getIterator();
+        $stmt->closeCursor();
+
+        $stmt = $this->connection->executeCacheQuery($query, [], [], $qcp);
+
+        $iterator = $stmt->getIterator();
+
+        self::assertCount(1, $iterator);
```

Comment:
This is not a valid behavior to expect. The fact that the iterator internally calls `fetchAll()` is a bug (see https://github.com/doctrine/dbal/pull/2718) and shouldn't be relied upon to expect that the statement is cached just because an iterator over its results was instantiated.